src/main/java/transforms/PaddingTransform.java

```diff
@@ -0,0 +1,114 @@
+package transforms;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.openlattice.shuttle.transformations.Transformation;
+import com.openlattice.shuttle.transformations.Transformations;
+import com.openlattice.shuttle.util.Cached;
+import com.openlattice.shuttle.util.Constants;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+
+public class PaddingTransform extends Transformation<Map<String, String>> {
+    private final String pattern;
+    private final int length;
+    private final boolean pre;
+    private final boolean cutoff;
+
+    /**
+     * A transform that pads the input string to a certain length by appending or prepending a repeating pattern
+     *
+     * @param pattern:            pad will be constructed by repeating this string
+     * @param length:             target length of resulting string
+     * @param pre:                whether to prepend (as opposed to append)
+     * @param cutoff:             whether to trim the string to the target size if it's larger
+     *
+     */
+    @JsonCreator
+    public PaddingTransform(
+            @JsonProperty( Constants.PATTERN ) String pattern,
+            @JsonProperty( Constants.LENGTH ) int length,
+            @JsonProperty( Constants.PRE ) boolean pre,
+            @JsonProperty( Constants.CUTOFF ) boolean cutoff
+            ) {
+        this.pattern = pattern;
+        this.length = length;
+        this.pre = pre;
+        this.cutoff = cutoff;
+    }
+
+    @JsonCreator
```

Comment:
You can't have two @JsonCreator's for the same class.  This doesn't run.
Either:
- make cutoff argument optional: 
```
this.cutoff = cutoff.orElse(false)
```
- just remove the jsoncreator in the second constructor.